packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core';
+import { EInputFieldType, IErrorMsgs } from './text-field.types';
+
+@Component({
+	tag: 'kv-text-field',
+	styleUrl: 'text-field.scss',
+	shadow: true,
+})
+export class KvTextField {
+	/** (optional) Text field tyoe */
+	@Prop({ reflect: true }) type!: EInputFieldType;
+	/** (optional) Text field label */
+	@Prop({ reflect: true }) label: string;
+	/** (optional) Text field input name */
+	@Prop({ reflect: true }) inputName: string;
+	/** (optional) Text field place holder */
+	@Prop({ reflect: true }) placeholder: string;
+	/** (optional) Text field disabled */
+	@Prop({ reflect: true }) disabled = false;
+	/** (optional) Text field required */
+	@Prop({ reflect: true }) required = false;
+	/** (optional) Text field loading state */
+	@Prop({ reflect: true }) loading = false;
+	/** (optional) Text field help text */
+	@Prop({ reflect: true }) helpText: string;
+	/** (optional) Text field error messages */
+	@Prop({ reflect: true }) errorMessages: IErrorMsgs;
+	/** (optional) Text field value */
+	@Prop({ reflect: true }) value: string;
+
+	/** Emitted when text field's value changes */
+	@Event() textChange: EventEmitter<InputEvent>;
+
+	private onTextChange = (event: InputEvent) => {
+		if (!event) {
+			event.preventDefault();
+		}
+		this.textChange.emit(event);
+	}
+
+	render() {
+		return (
+			<Host>
+				<div class="text-field-container">
+					{this.required && <span class="error">
+						*
+					</span>}
+					{this.label && <span class="label">
+						{this.label}
+					</span>}
+					<div class="input-container">
+						{
+							(this.type !== EInputFieldType.Number &&
```

Comment:
I don't understand this.
Firstly I encourage you to create a variable with this condition and reuse it here to bring more clarity to the `JSX` code.

Then you are checking if the type is not number and it's not loading and showing the input if this condition is met. 
What if the input type is a number? Right now it's not being shown at all.

I think what you intend to do is check if the type is a number and apply a regexp pattern to it.

Reply:
It is for when are number use another type of input with mask but for now we will not have this and I will remove.